analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocation.cs

```diff
@@ -884,12 +884,12 @@ public void Invocation_TargetMethodIsDelegateInvoke()
         }
 
         [DataTestMethod]
-        [DataRow("null", "null", true, "Null", "Null")]
-        [DataRow("null", "new object()", false, "Null", "NotNull")]
-        [DataRow("new object()", "null", false, "NotNull", "Null")]
-        [DataRow("new int?()", "null", false, "NotNull", "Null")]   // Should be "true" and "Null" because new int()? is still null
-        [DataRow("new int?(42)", "null", false, "NotNull", "Null")]
-        public void Invocation_Equals_LearnResult(string left, string right, bool? expectedResult, string expectedConstraintsLeft, string expectedConstraintsRight)
+        [DataRow("null", "null", true, new[] { ConstraintKind.ObjectNull }, new[] { ConstraintKind.ObjectNull })]
+        [DataRow("null", "new object()", false, new[] { ConstraintKind.ObjectNull }, new[] { ConstraintKind.ObjectNotNull })]
+        [DataRow("new object()", "null", false, new[] { ConstraintKind.ObjectNotNull }, new[] { ConstraintKind.ObjectNull })]
+        [DataRow("new int?()", "null", false, new[] { ConstraintKind.ObjectNotNull }, new[] { ConstraintKind.ObjectNull })]   // Should be "true" and "Null" because new int()? is still null
+        [DataRow("new int?(42)", "null", false, new[] { ConstraintKind.ObjectNotNull }, new[] { ConstraintKind.ObjectNull })]
+        public void Invocation_Equals_LearnResult(string left, string right, bool expectedResult, ConstraintKind[] expectedConstraintsLeft, ConstraintKind[] expectedConstraintsRight)
```

Comment:
As long as we expect exactly one, we don't need to complicated the syntax with arrays above. It can be the assertion below (.ContainsSingle?)
```suggestion
        public void Invocation_Equals_LearnResult(string left, string right, bool expectedResult, ConstraintKind expectedConstraintsLeft, ConstraintKind expectedConstraintsRight)
```